actionpack/lib/action_dispatch/journey/formatter.rb

```diff
@@ -57,6 +57,9 @@ def message
       end
 
       def generate(name, options, path_parameters)
+        bind_params = options.delete(:bind_params) || {}
+        original_options = options.dup
```

Comment:
Why `dup` here? You are not mutating the content of `options`, actually you are with the `delete` on top, so if you are trying to avoid mutating the argument the `dup` needs to be before the `delete`.

Reply:
I moved the dup before the delete (which allows me to remove the other dup later one)